registry/storage/linkedblobstore.go

```diff
@@ -74,6 +75,20 @@ func (lbs *linkedBlobStore) Put(ctx context.Context, mediaType string, p []byte)
 		return distribution.Descriptor{}, err
 	}
 
+	// Remove a tombstone if one exists for this blob.  Continue to write
+	// the blob as it may have been removed by the GC sweep process.  If
+	// deletes are not enabled, fall through
+	dgst, err := digest.FromBytes(p)
```

Comment:
linkedBlobStore uses a tomb for Put (to remove an existing tombstone) and Delete (to place a tombstone, and invalidate cache descriptor entries (via the linkedBlobStatter) ).

linkedBlobStatter uses a tomb for Stat (to return false if there is a tombstone).  It doesn't mutate tombstones.